File: HW1_Add&Max/addAndMax.py
def addAndMax(A,B):
    import numpy as np
    n1, n2 = np.size(A,1),  np.size(B,1)
    m1, m2 = np.size(A,0),  np.size(B,0)
    m, n = max(m1, m2), max(n1, n2)
    C = np.zeros((m,n))
    max_num = C[0][0]
    for i in range(m):
        for j in range(n):
            if i < m1 and i < m2 and j < n1 and j < n2:
                C[i][j] = (A[i][j] + B[i][j])
            elif i < m2 and j < n2 and (j >= n1 or i >= m1):
                C[i][j] = (B[i][j])
            elif i < m1 and j < n1 and (i >= m2 or j >= n2) :
                C[i][j] = A[i][j]
            else:
                C[i][j] = 0
            if max_num < C[i][j]:
                max_num = C[i][j]
    ans = max_num
    return ans

File: HW1_Add&Max/addAndMax.m
function output = addAndMax(A,B)
[m1 n1] = size(A);
[m2 n2] = size(B);
if m1 < m2
    m = m2;
else
    m = m1;
end
if n1 < n2
    n = n2;
else
    n = n1;
end
C = zeros(m,n);

for i = 1:m
    for j = 1:n
        if i <= m1 & i <= m2 & j <= n1 & j <= n2
            C(i,j) = A(i,j) + B(i,j);
        elseif i > m1 & j <= n2
            C(i,j) = B(i,j);
        elseif j > n2 & i <= m1
            C(i,j) = A(i,j);
        else
            C(i,j) = 0;
        end
    end
end

max = C(1);
for i = 1:m*n
    if max < C(i)
        max = C(i);
    end
end
output = max;